addons/sys_gestures/XEH_postInit.sqf

```diff
@@ -0,0 +1,70 @@
+#include "script_component.hpp"
+if (is3DEN || {!hasInterface || {!(GVAR(gesturesEnabled))}}) exitWith {};
+if (!isClass (configFile >> "CfgPatches" >> "ace_common")) exitWith {}; // No ACE exit
+
+GVAR(vestRadioArr) = parseSimpleArray (GVAR(vestRadios));
+GVAR(headsetRadioArr) = parseSimpleArray (GVAR(headsetRadios));
+
+if (GVAR(stopADS)) then {
+    GVAR(disallowedViews) = ["GROUP"];
+};
+
+["acre_startedSpeaking", {
+    params ["_unit", "_onRadio", "_radio"];
+
+    if (!_onRadio ||
+        { !isNull objectParent _unit } ||
+        { cameraView in GVAR(disallowedViews) } ||
+        { ace_common_isReloading } ||
+        { isWeaponDeployed _unit } ||
+        { animationState _unit in GVAR(blackListAnims) } ||
+        { currentWeapon _unit in GVAR(binoClasses) } ) exitWith {};
+
+    private _hasVest = vest _unit != "";
+    private _hasHeadgear = headgear _unit != "";
+    if (!_hasVest && !_hasHeadgear) exitWith {};
+
+    private _baseRadio = _radio call acre_api_fnc_getBaseRadio;
+    private _isVestRadio = _baseRadio in GVAR(vestRadioArr);
+    private _isHeadsetRadio = _baseRadio in GVAR(headsetRadioArr);
+
+    if (_hasVest && _isVestRadio) then {
+        _unit playActionNow ([QGVAR(vest), QGVAR(vest_noADS)] select GVAR(stopADS));
+        _unit setVariable [QGVAR(onRadio), true];
+    };
+
+    if (_hasHeadgear && _isHeadsetRadio) then {
+        _unit playActionNow ([QGVAR(helmet), QGVAR(helmet_noADS)] select GVAR(stopADS));
+        _unit setVariable [QGVAR(onRadio), true];
+    };
+}] call CBA_fnc_addEventHandler;
+
+["acre_stoppedSpeaking", {
+    params ["_unit", "_onRadio"];
+
+    if (!_onRadio) exitWith {};
+
+    // If the unit started a reload while already talking, need to wait to finish to not delete a magazine
+    [
+        {!ace_common_isReloading},
+        {
+            // Wait 1 frame as mag doesn't report as loaded til events completed
+            [FUNC(stopGesture), _this] call CBA_fnc_execNextFrame;
+        },
+        _unit
+    ] call CBA_fnc_waitUntilAndExecute;
+}] call CBA_fnc_addEventHandler;
+
+acre_player addEventHandler ["GetInMan", {
```

Comment:
won't handle unit switching (zeus rc)

Reply:
Is there a cba event for unit switched or anything we can hook?

Reply:
```sqf
["unit", {
    params ["_newUnit", "_oldUnit"];
    systemChat str "Unit changed.";
}, true] call CBA_fnc_addPlayerEventHandler;
```

Reply:
Should be handled now, will need a quick test to ensure it works properly.

Reply:
This still won't work. Event gets added to `acre_player` and not to the possibly remote controlled units. Problem isn't stopping gesture when switching, problem is switching and then getting in on that unit. CBA has a Player EH `"vehicle"` as well. CBA even has a bonus of supporting scripted moves.

Same for the `"WeaponDeployed"`, but not sure if any CBA Player EH handles that.